components/detournavigator/navmeshmanager.hpp

```diff
@@ -22,6 +22,15 @@ namespace DetourNavigator
     public:
         NavMeshManager(const Settings& settings);
 
+        ~NavMeshManager()
+        {
+            mAsyncNavMeshUpdater.stop();
+            mCache.clear();
+            mChangedTiles.clear();
+            mPlayerTile.clear();
+            mLastRecastMeshManagerRevision.clear();
+        }
```

Comment:
IIRC, a class's members are constructed in the order of declaration, and destructed in reverse order. So as long as `mAsyncNavMeshUpdater` is the last declared data member, it will be destructed first before `mCache` and the others.

Reply:
The idea is th `mAsyncNavMeshUpdater` **is not** the last, and attempt to just move it leads to crashes on exit.

Reply:
Okay, fair enough.